gnovm/pkg/gnolang/nodes.go

```diff
@@ -385,33 +390,65 @@ type BasicLitExpr struct {
 	Value string
 }
 
+func (x *BasicLitExpr) addressability() Addressability {
+	return AddressabilityUnsatisfied
+}
+
 type BinaryExpr struct { // (Left Op Right)
 	Attributes
 	Left  Expr // left operand
 	Op    Word // operator
 	Right Expr // right operand
 }
 
+func (x *BinaryExpr) addressability() Addressability {
+	return AddressabilityUnsatisfied
+}
+
 type CallExpr struct { // Func(Args<Varg?...>)
 	Attributes
-	Func    Expr  // function expression
-	Args    Exprs // function arguments, if any.
-	Varg    bool  // if true, final arg is variadic.
-	NumArgs int   // len(Args) or len(Args[0].Results)
+	Func           Expr  // function expression
+	Args           Exprs // function arguments, if any.
+	Varg           bool  // if true, final arg is variadic.
+	NumArgs        int   // len(Args) or len(Args[0].Results)
+	Addressability Addressability
+}
+
+func (x *CallExpr) addressability() Addressability {
+	return x.Addressability
 }
 
 type IndexExpr struct { // X[Index]
 	Attributes
-	X     Expr // expression
-	Index Expr // index expression
-	HasOK bool // if true, is form: `value, ok := <X>[<Key>]
+	X              Expr // expression
+	Index          Expr // index expression
+	HasOK          bool // if true, is form: `value, ok := <X>[<Key>]
+	Addressability Addressability
+}
+
+func (x *IndexExpr) addressability() Addressability {
+	// In this case NotApplicable means that it wasn't set, the default value.
+	if x.Addressability == AddressabilityNotApplicable {
```

Comment:
When does this happen?

Reply:
This happens when it is not set in TRANS_LEAVE, here https://github.com/gnolang/gno/blob/804c17e0d7d15e459ac2430fb80d585545780903/gnovm/pkg/gnolang/preprocess.go#L1500-L1506

I've added a clarifying comment. https://github.com/gnolang/gno/pull/2731/commits/44cc298b03619832255641603d5575d94979fcae